### ISSUE TITLE
Bitcoin balance disappears from the UI

### ISSUE THREAD

**Issue by zathras-crypto** [OmniLayer/omnicore#219]
There is an issue where the BTC balance can disappear from the UI.

@CraigSellars mentioned to me that he had seen his BTC balance disappear in omnicore-qt when running on testnet.  Nothing had actually changed balance-wise, but the overview page just flipped to showing 0 BTC in the wallet.

![btcdisappears](https://cloud.githubusercontent.com/assets/5633351/9649745/4475d050-523c-11e5-803a-541a0a5d3d36.png)

I can see how this would raise significant concern in users on mainnet!

I'd been having a hard time replicating it, and now I believe I've found the cause I see why hehe - basically the symptoms only occur when a new longer chain comes along and orphans the old one so the state is rolled back.  

When the state is rolled back, we pass through a signal to the UI (```OmniStateInvalidated```) which causes the UI to reset and data to be repopulated.

The problem is we reset *everything*, and then repopulate *everything for the Omni layer*.  That means we reset (zero) the Bitcoin balance too, but it's not part of the Omni layer so the data doesn't get repopulated.

Thus the resulting issue - when there is a chain reorg the user will see their BTC balance drop to zero.

Thankfully now I know what the problem is it's a nice straightforward fix (we just need to request the BTC balance data from the ```walletModel``` and update the overview accordingly from ```OverviewPage::reinitOmni()```).  

I've submitted a PR for the fix here https://github.com/OmniLayer/omnicore/pull/220

**Comment by dexX7** [comment]
Great catch! I'm a bit surprised, because, if I'm not mistaken, then the Core UI is updated frequently by a timer, but appearingly not, in this case.

**Comment by dexX7** [comment]
Confirmed.

I started 2x QT (with different ports and datadirs), but didn't connect them. In one I mined 101 blocks, in the other 120+ blocks. Then I connected them and the first node shows 0.00 BTC, as described.

**Comment by zathras-crypto** [comment]
> Did you actually test this @zathras-crypto?

I did indeed :)

It's pretty easy to replicate, just take the tip and leave it running on testnet, there are frequent reorgs and it won't be long before you see the behaviour occur.  The BTC drops to zero and stays there.

> then the Core UI is updated frequently by a timer

That's what I thought too, but I added a whole bunch of extra debugging to show me where the balances were being set, I could clearly see the BTC balance being zero'd when ```reinitOmni()``` was called, and then there was no subsequent update to set it back.

> I started 2x QT (with different ports and datadirs), but didn't connect them. In the first one I mined 101 blocks, in the second one 120. Then I send 1 BTC from the second to an address of the first, mined a block with each. Finally I connected them.

Perhaps your BTC send with the final block is triggering a balance update - try populating any balances before triggering the chain rollback perhaps?
